cmake/SetupPch.cmake

```diff
@@ -206,10 +210,11 @@ if (USE_PCH)
     )
 
   # Create the ${SPECTRE_PCH} that libraries and executables will depend on
-  add_library(${SPECTRE_PCH} INTERFACE)
-  target_link_libraries(
+  add_custom_target(
```

Comment:
I think it was this change that broke things, but I don't have time to investigate